chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/CertifiedProductController.java

```diff
@@ -188,6 +189,24 @@
 			} catch(EntityRetrievalException ex) {}
 		}
 		
+		List<CertificationCriterionEntity> retiredTestTools = cpDao.getRetiredTestTools(existingProduct.getId());
+		Boolean hasRetiredTestTool = false;
+		if(retiredTestTools.size() > 0){
+			hasRetiredTestTool = true;
+		}
+		
+		Boolean updateIcs = updateRequest.getIcs();
+		Boolean existingProductIcs = existingProduct.getIcs();
+		if(updateIcs == false && existingProductIcs == true && hasRetiredTestTool == true){
+			List<String> retiredTestToolNumbers = new ArrayList<String>();
+			for(CertificationCriterionEntity cce : retiredTestTools){
+				retiredTestToolNumbers.add(cce.getNumber());
+			}
+			
+			updateRequest.getErrorMessages().add("Cannot set Ics to false for a Certified Product with Ics=true and attested criteria that have a retired Test Tool. "
+					+ "The following are attested criteria have a retired Test Tool: " + retiredTestToolNumbers.toString());

```

Comment:
"The following are attested criteria have" -> "The following are attested criteria *that* have"